### PR TITLE
[types@next-auth] - Update client.d.ts

### PR DESCRIPTION
**Pull request by StefanSelfTaught**
Added redirect type to signIn and signOut

Please fill in this template.

- [ ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/next-auth/client.d.ts [hunk 1]
```diff
@@ -50,10 +50,16 @@ declare function signin(
     provider?: string,
     data?: GenericObject & {
         callbackUrl?: string;
+        redirect?: boolean;
     },
 ): Promise<void>;
 declare const signIn: typeof signin;
-declare function signout(data?: { callbackUrl?: string }): Promise<void>;
+declare function signout(
+    data?: {
```
Comment (lluia) [inline types/next-auth/client.d.ts:58]: There's a trailing white-space on this line 📐 , which you'll need to remove for the lint process [not to complain on the CI](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/51477/checks?check_run_id=1999784269) 👍🏼

File: types/next-auth/client.d.ts [hunk 2]
```diff
@@ -50,10 +50,16 @@ declare function signin(
...
         callbackUrl?: string;
+        redirect?: boolean;
     },
 ): Promise<void>;
 declare const signIn: typeof signin;
-declare function signout(data?: { callbackUrl?: string }): Promise<void>;
+declare function signout(
+    data?: { 
+        callbackUrl?: string; 
+        redirect?: boolean;
```
Comment (lluia) [inline types/next-auth/client.d.ts:60]: Let's [add a test](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/4bea1bd982be671edf210b7010aba4ce0f627bd1/types/next-auth/next-auth-tests.ts#L265) to verify this new option 🪴

**Review by 0ubbe — CHANGES_REQUESTED** [review]
Thanks, small two 💬  to make the CI pass 👍🏽

**Review by 0ubbe — APPROVED** [review]
Thanks 💚

**Comment by StefanSelfTaught** [comment]
Ready to merge

**Pull request merged**